src/main/java/bwapi/Unit.java

```diff
@@ -835,14 +835,17 @@ public UnitType getBuildType() {
      * @see #isTraining
      */
     public List<UnitType> getTrainingQueue() {
-        return IntStream.range(0, getTrainingQueueCount())
-                .mapToObj(i -> game.isLatComEnabled() && self().trainingQueue[i].valid(game.getFrameCount()) ?
-                        self().trainingQueue[i].get() :
-                        UnitType.idToEnum[unitData.getTrainingQueue(i)])
-                .collect(Collectors.toList());
+        return IntStream.range(0, getTrainingQueueCount()).mapToObj(this::getTrainingQueueAt).collect(Collectors.toList());
+    }
+
+    public UnitType getTrainingQueueAt(int i) {
+        if (game.isLatComEnabled() && self().trainingQueue[i].valid(game.getFrameCount())) {
+            return self().trainingQueue[i].get();
+        }
+        return UnitType.idToEnum[unitData.getTrainingQueue(i)];
     }
 
-    int getTrainingQueueCount() {
+    public int getTrainingQueueCount() {
```

Comment:
Two new methods `getTrainingQueueAt()` and `getTrainingQueueCount()` allow peeking at the training queue without creating a new collection each time.

Reply:
I like it, can you add javadoc similar to the other methods? I think we should also mention in those docs that they are not exposed by native BWAPI